In [1]:
import csv

with open("guns.csv") as f:
    data = list(csv.reader(f))
    
headers = data[0]
    
headers

['',
 'year',
 'month',
 'intent',
 'police',
 'sex',
 'age',
 'race',
 'hispanic',
 'place',
 'education']

In [2]:
data = data[1:]
data[:5]

[['1',
  '2012',
  '01',
  'Suicide',
  '0',
  'M',
  '34',
  'Asian/Pacific Islander',
  '100',
  'Home',
  '4'],
 ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'],
 ['3',
  '2012',
  '01',
  'Suicide',
  '0',
  'M',
  '60',
  'White',
  '100',
  'Other specified',
  '4'],
 ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4'],
 ['5',
  '2012',
  '02',
  'Suicide',
  '0',
  'M',
  '31',
  'White',
  '100',
  'Other specified',
  '2']]

In [3]:
years = [year[1] for year in data]

year_counts = {}

for year in years:
    if year not in year_counts:
        year_counts[year] = 1
    else:
        year_counts[year] += 1
        
year_counts

{'2012': 33563, '2013': 33636, '2014': 33599}

In [4]:
import datetime

dates = [datetime.datetime(int(day[1]), int(day[2]), 1) for day in data]
dates[:5]

[datetime.datetime(2012, 1, 1, 0, 0),
 datetime.datetime(2012, 1, 1, 0, 0),
 datetime.datetime(2012, 1, 1, 0, 0),
 datetime.datetime(2012, 2, 1, 0, 0),
 datetime.datetime(2012, 2, 1, 0, 0)]

In [5]:
date_counts = {}

for day in dates:
    if day not in date_counts:
        date_counts[day] = 1
    else:
        date_counts[day] += 1
        
date_counts

{datetime.datetime(2012, 1, 1, 0, 0): 2758,
 datetime.datetime(2012, 2, 1, 0, 0): 2357,
 datetime.datetime(2012, 3, 1, 0, 0): 2743,
 datetime.datetime(2012, 4, 1, 0, 0): 2795,
 datetime.datetime(2012, 5, 1, 0, 0): 2999,
 datetime.datetime(2012, 6, 1, 0, 0): 2826,
 datetime.datetime(2012, 7, 1, 0, 0): 3026,
 datetime.datetime(2012, 8, 1, 0, 0): 2954,
 datetime.datetime(2012, 9, 1, 0, 0): 2852,
 datetime.datetime(2012, 10, 1, 0, 0): 2733,
 datetime.datetime(2012, 11, 1, 0, 0): 2729,
 datetime.datetime(2012, 12, 1, 0, 0): 2791,
 datetime.datetime(2013, 1, 1, 0, 0): 2864,
 datetime.datetime(2013, 2, 1, 0, 0): 2375,
 datetime.datetime(2013, 3, 1, 0, 0): 2862,
 datetime.datetime(2013, 4, 1, 0, 0): 2798,
 datetime.datetime(2013, 5, 1, 0, 0): 2806,
 datetime.datetime(2013, 6, 1, 0, 0): 2920,
 datetime.datetime(2013, 7, 1, 0, 0): 3079,
 datetime.datetime(2013, 8, 1, 0, 0): 2859,
 datetime.datetime(2013, 9, 1, 0, 0): 2742,
 datetime.datetime(2013, 10, 1, 0, 0): 2808,
 datetime.datetime(2013, 11,

In [6]:
sex_counts = {}

for sex in data:
    if sex[5] not in sex_counts:
        sex_counts[sex[5]] = 1
    else:
        sex_counts[sex[5]] += 1
        
sex_counts

{'F': 14449, 'M': 86349}

In [7]:
race_counts = {}

for race in data:
    if race[7] not in race_counts:
        race_counts[race[7]] = 1
    else:
        race_counts[race[7]] += 1
        
race_counts

{'Asian/Pacific Islander': 1326,
 'Black': 23296,
 'Hispanic': 9022,
 'Native American/Native Alaskan': 917,
 'White': 66237}

# Gun Deaths

Significantly more blacks and whites are killed every year as a results of firearms. Further multi-category analysis would be beneficial in showing other trends.

- __Examine Race/Sex/Intent/Police and compare with population ratios within the U.S. to identify propensity amongst various populations__

- __Visualize the results__

<br>

<br>

>**Data**

> - {'Asian/Pacific Islander': 1326, 'Black': 23296, 'Hispanic': 9022, 'Native American/Native Alaskan': 917, 'White': 66237}

> - {'F': 14449, 'M': 86349}

In [8]:
with open("census.csv") as f:
    census = list(csv.reader(f))
    
census

[['Id',
  'Year',
  'Id',
  'Sex',
  'Id',
  'Hispanic Origin',
  'Id',
  'Id2',
  'Geography',
  'Total',
  'Race Alone - White',
  'Race Alone - Hispanic',
  'Race Alone - Black or African American',
  'Race Alone - American Indian and Alaska Native',
  'Race Alone - Asian',
  'Race Alone - Native Hawaiian and Other Pacific Islander',
  'Two or More Races'],
 ['cen42010',
  'April 1, 2010 Census',
  'totsex',
  'Both Sexes',
  'tothisp',
  'Total',
  '0100000US',
  '',
  'United States',
  '308745538',
  '197318956',
  '44618105',
  '40250635',
  '3739506',
  '15159516',
  '674625',
  '6984195']]

In [9]:
mapping = {
    'Asian/Pacific Islander': int(census[1][14]) + int(census[1][15]), 
    'Black': int(census[1][12]), 
    'Hispanic': int(census[1][11]), 
    'Native American/Native Alaskan': int(census[1][13]), 
    'White': int(census[1][10])
}

mapping

{'Asian/Pacific Islander': 15834141,
 'Black': 40250635,
 'Hispanic': 44618105,
 'Native American/Native Alaskan': 3739506,
 'White': 197318956}

In [10]:
race_per_hundredk = {}

for key in race_counts:
    if key not in race_per_hundredk:
        race_per_hundredk[key] = race_counts[key] / mapping[key]
        race_per_hundredk[key] = race_per_hundredk[key] * 100000

race_per_hundredk
    

{'Asian/Pacific Islander': 8.374309664161762,
 'Black': 57.8773477735196,
 'Hispanic': 20.220491210910907,
 'Native American/Native Alaskan': 24.521955573811088,
 'White': 33.56849303419181}

# Gun Deaths by Race per 100k
<br>
<br>
-  Asian/Pacific Islander: 8.374309664161762
-  Black: 57.8773477735196
-  Hispanic: 20.220491210910907
-  Native American/Native Alaskan: 24.521955573811088
-  White: 33.56849303419181


In [20]:
intents = [row[3] for row in data]
races = [row[7] for row in data]

homicide_race_counts = {}

for i, race in enumerate(races):
    if intents[i] == "Homicide":
        if race not in homicide_race_counts:
            homicide_race_counts[race] = 1
        else:
            homicide_race_counts[race] += 1
            
homicide_race_counts


{'Asian/Pacific Islander': 559,
 'Black': 19510,
 'Hispanic': 5634,
 'Native American/Native Alaskan': 326,
 'White': 9147}

In [22]:
homicide_race_per_100k = {}

for key in homicide_race_counts:
    if key not in homicide_race_per_100k:
        homicide_race_per_100k[key] = homicide_race_counts[key] / mapping[key]
        homicide_race_per_100k[key] = homicide_race_per_100k[key] * 100000
        
        
homicide_race_per_100k

{'Asian/Pacific Islander': 3.530346230970155,
 'Black': 48.471284987180944,
 'Hispanic': 12.627161104219914,
 'Native American/Native Alaskan': 8.717729026240365,
 'White': 4.6356417981453335}

# Homicides by Race per 100k of Population
<br>
> ### Asian/Pacific Islander: 3.530346230970155
> ### Black: 48.471284987180944
> ### Hispanic: 12.627161104219914
> ### Native American/Native Alaskan: 8.717729026240365
> ### White: 4.6356417981453335

# TO DO
<br>
- [ ] Examine Intent Data
- [ ] Examine and compare Education
- [ ] Visualize Data